src/Shared/AWS/AWSSemanticConventions.cs

```diff
@@ -0,0 +1,329 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.ComponentModel;
+using System.Diagnostics;
+
+// disable Style Warnings to improve readability of this specific file.
+#pragma warning disable SA1124
+#pragma warning disable SA1005
+#pragma warning disable SA1514
+#pragma warning disable SA1201
+#pragma warning disable SA1623
+
+namespace OpenTelemetry.AWS;
+
+/// <summary>
+/// Abstracts the complexities of honoring <see cref="AWS.SemanticConventionVersion"/>.
+/// <para/>
+/// Classes emitting attributes can use the extension methods in this class to build
+/// a List of <see cref="KeyValuePair{K,V}"/>s containing
+/// Attribute Name and Value without  needing to know which version of the
+/// Semantic Convention to use.
+/// <example>
+/// Below is a hypothetical example showing how attributes can be constructed.  It is not necessary
+/// for this consumer to accommodate differing behavior based on <see cref="SemanticConventionVersion"/>,
+/// as that will be handled by the extension methods (ie <see cref="AddAttributeCloudResourceId{T}"/>) themselves.
+/// <code>
+/// <![CDATA[
+/// var resourceAttributes =
+///     new List<KeyValuePair<string, object>>()
+///         .AddAttributeCloudResourceId(containerArn)
+///         .AddAttributeFoo("value 1")
+///         .AddAttributeBar("value 2");
+/// ]]>
+/// </code>
+/// </example>
+/// </summary>
+internal static partial class AWSSemanticConventions
+{
+    /// <summary>
+    /// <para>
+    /// Sets the <see cref="SemanticConventionVersion"/> that will be used to resolve attribute names.
+    /// </para>
+    /// This should be set by an Options class.
+    /// </summary>
+    public static SemanticConventionVersion SemanticConventionVersion { get; set; } = DefaultSemanticConventionVersion;
+
+    /// <summary>
+    /// Options classes should use this value for initialization.
+    /// <para />
+    /// Example:
+    /// <code>
+    /// <![CDATA[
+    /// public class AWSLambdaInstrumentationOptions
+    /// {
+    ///    /// <inheritdoc cref="OpenTelemetry.AWS.SemanticConventionVersion"/>
+    ///    public SemanticConventionVersion SemanticConventionVersion { get; set; } = AWSSemanticConventions.DefaultSemanticConventionVersion;
+    /// }
+    /// ]]>
+    /// </code>
+    /// </summary>
+    internal const SemanticConventionVersion DefaultSemanticConventionVersion = SemanticConventionVersion.Latest;
```

Comment:
At least for one release, wouldn't it make sense to choose the default such that behavior remains unchanged and users have the option to upgrade at their own convenience?

Reply:
I think we should always set `DefaultSemanticConventionVersion` to a specific version rather than `SemanticConventionVersion.Latest` and only update it when we introduce a new major version. This way users can safely upgrade minor versions without facing breaking changes.

That said, after glancing over other packages, it seems that none handle backward compatibility for semantic convention versioning. So I'm find with keeping it this way if we are ok with making maintaining backward compatibility a user responsibly using the configuration that was added.

Reply:
This request makes sense to me - we can make this more conservative and set the default to the 1.27, which mostly aligns to what the current library uses.  

This way users wont see any changes in Semantic Convention unless they opt-in.

I've made this update.